### DISCUSSION TITLE
Add an option to install Roon Bridge?

### DISCUSSION THREAD

**Discussion by wjanssen** [micro-nova/AmpliPi#78] [Ideas]
[Roon](https://roonlabs.com/) is a widely used music player software which streams locally from a "core" on the LAN to audio devices that support network access.  It can be supported on audio equipment, include Pi-based devices like AmpliPi, by downloading and installing the Linux version of [Roon Bridge](https://help.roonlabs.com/portal/en/kb/articles/roonbridge).  It would be great if the AmpliPi web page had a button which would cause it to [download and install Roon Bridge](https://help.roonlabs.com/portal/en/kb/articles/linux-install).

**Comment by linknum23** [comment]
So as far as I can tell AmpliPi could easily add at least one Roon bridge, potentially up to 4 since AmpliPi has 4 independent stereo channels. One possible complication is that a Roon bridge can switch between stereo channels independently of the AmpliPi unit which could cause conflicts ( such as multiple songs playing out the same speakers/zone ) if the AmpliPi is playing additional audio.

Is your plan to use the AmpliPi and control it entirely using a Roon "core"? If so how many different audio streams would you play at once? Do you expect to be able to control zone volumes individually using the Roon "core"?

I only ask because adding a bridge that can play audio will probably be very easy, but tightly integrating AmpliPi with the Roon system might be significantly more complicated.

**Comment by linknum23** [comment]
I looked into this a little further and it seems like the easiest way to integrate with Roon currently is to use AmpliPi's existing airplay interface. AmpliPi can play 4 different airplay streams at once, each connected to one or multiple zones.

**Comment by linknum23** [comment]
Looking into this further, I don't think adding a Roon Bridge makes much sense initially, since the use case of the Roon Bridge is covered by the existing Airplay interface. 

Full Roon integration, at a "Roon Ready" level would likely involve porting the whole AmpliPi software system to Roon's JS framework. This would be outside the scope of our goals for software, but since the whole system is open it could be taken up by the community with our assistance. @wjanssen do you know of anyone from the Roon community be interested in adding that support?

**Comment by linknum23** [comment]
Closing this for now, since we haven't really gotten any interest in this feature. Feel free to open it back up.

**Comment by wjanssen** [comment]
I think this got conflated with "Roon Support", which is a much more involved and twisty ask.

Just to clarify: "Roon Ready" is a fairly heavyweight commitment which would include certification from Roon Labs.  It involves signing an NDA and getting the SDK from Roon, and hooking up the SDK correctly to your software, then sending a unit to Roon Labs for testing and certification.

"Roon Bridge", on the other hand, is a drop-in software component that doesn't require tinkering with your software.  It uses the standard audio frameworks for the OS you install it on (I presume this would be ALSA for your presumably Linux platform) to see what the audio capabilities are and manipulate them.

My suggestion was that you add a button which would cause your system to download and install Roon Bridge.  This would automatically make your system visible to a Roon Core running elsewhere on the LAN.  How Roon would "see" your system is an interesting question, one perhaps most easily answered simply by downloading Roon Bridge and installing it, to see what happens.  If it works, great!  If not, this would become a more labor-intensive sub-project.

**Reply by linknum23** [reply]
Is there a way we can test Roon bridge in the Roon ecosystem without buying anything? I have a feeling it will have the issues discussed above but I am willing to try it out.

**Comment by mwbrink** [comment]
I use Roon and would like to cast my vote to add Roon Bridge.
